ui/app/AppLayouts/Wallet/views/RecipientView.qml

```diff
@@ -0,0 +1,193 @@
+
+import StatusQ 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Backpressure 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1 as StatusQUtils
+
+import AppLayouts.Wallet 1.0
+
+import shared.controls 1.0 as SharedControls
+import shared.stores.send 1.0
+import shared.popups.send.panels 1.0
+import shared.popups.send 1.0
+import shared.popups.send.controls 1.0
+
+import utils 1.0
+
+Loader {
+    id: root
+
+    required property var savedAddressesModel
+    required property var myAccountsModel
+
+    property string selectedRecipientAddress
+    property int selectedRecipientType: Helpers.RecipientAddressObjectType.Address
+    property bool interactive: true
+
+    signal resolveENS(string ensName, string uuid)
+
+    function ensNameResolved(resolvedPubKey, resolvedAddress, uuid) {
+        if(uuid !== d.uuid) {
+            return
+        }
+        root.selectedRecipientAddress = resolvedAddress
+    }
+
+    QtObject {
+        id: d
+
+        property bool isValidAddress: true
+        property bool isBeingEvaluated: false
+
+        property string uuid
+
+        readonly property var validateInput: Backpressure.debounce(root, 500, function (address) {
+            d.isValidAddress = Utils.isValidAddress(address)
+            const isENSName = Utils.isValidEns(address)
+
+            if(d.isValidAddress) {
+                root.selectedRecipientAddress = address
+                d.isBeingEvaluated = false
+            }
+            else if(isENSName) {
+                d.uuid = Utils.uuid()
+                return root.resolveENS(address, uuid)
+            } else {
+                root.selectedRecipientAddress = ""
+                d.isBeingEvaluated = false
+            }
+        })
+
+        readonly property var accountsSelectedEntry: ModelEntry {
+            sourceModel: root.myAccountsModel
+            key: "address"
+            value: root.selectedRecipientAddress
+        }
+
+        readonly property var savedAddrSelectedEntry: ModelEntry {
+            sourceModel: root.savedAddressesModel
+            key: "address"
+            value: root.selectedRecipientAddress
+
+        }
+
+        function clearValues() {
+            root.selectedRecipientAddress = ""
+            root.selectedRecipientType = Helpers.RecipientAddressObjectType.Address
+        }
+    }
+
+    sourceComponent: root.selectedRecipientType === Helpers.RecipientAddressObjectType.SavedAddress ?
+                         savedAddressRecipient:
+                         root.selectedRecipientType === Helpers.RecipientAddressObjectType.Account ?
+                             myAccountRecipient:
+                             root.selectedRecipientType === Helpers.RecipientAddressObjectType.RecentsAddress ?
+                                 recentsRecipient : addressRecipient
+
+    Component {
+        id: savedAddressRecipient
+        SavedAddressListItem {
+            implicitWidth: parent.width
```

Comment:
```suggestion
            width: ListView.view.width
```

Reply:
there is no ListView here, also this component will be reworked under https://github.com/status-im/status-desktop/issues/16916